Runtime/Scripts/BaseMeshSync.Modifiers.cs

```diff
@@ -0,0 +1,544 @@
+                        var s = new StringBuilder(arrayLength + 1); // +1 for string terminator
+                        msPropertyInfoCopyData(propertyInfoData.self, s);
+                        propertyValue = s.ToString();
+                        break;
+                    }
+
+                default:
+                    Debug.LogError($"Type {propertyInfoData.type} not implemented");
+                    break;
+            }
+        }
+
+        public PropertyInfoDataType type;
+        public PropertyInfoDataSourceType sourceType;
+
+        [field: SerializeField]
+        public string name { get; private set; }
+
+        [field: SerializeField]
+        public float min { get; private set; }
+
+        [field: SerializeField]
+        public float max { get; private set; }
+
+        [field: SerializeField]
+        public string path { get; private set; }
+
+        [field: SerializeField]
+        public string modifierName { get; private set; }
+
+        [field: SerializeField]
+        public string propertyName { get; private set; }
+
+        [field: SerializeField]
+        public int arrayLength { get; private set; }
+
+        // object cannot be serialized by unity, save a string representation of it:
+        public object propertyValue;
+        [SerializeField, HideInInspector]
+        private string propertyValueSerialized;
+
+        public string ID
+        {
+            get
+            {
+                return $"{path}_{name}";
+            }
+        }
+
+        public bool Matches(PropertyInfoDataWrapper other)
+        {
+            return
+                name == other.name &&
+                type == other.type &&
+                sourceType == other.sourceType &&
+                propertyName == other.propertyName &&
+                arrayLength == other.arrayLength;
+        }
+
+        static T[] ParseArray<T>(string propertyValueSerialized, Func<string, T> parse)
+        {
+            string[] v = propertyValueSerialized.Substring(2).Split('|');
+            var array = new T[v.Length];
+            for (int i = 0; i < v.Length; i++)
+            {
+                array[i] = parse(v[i]);
+            }
+
+            return array;
+        }
+
+        static string SaveArray<T>(string prefix, T[] array)
+        {
+            StringBuilder sb = new StringBuilder(prefix);
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (i < array.Length)
+                {
+                    sb.Append("|");
+                }
+
+                sb.Append(array[i]);
+            }
+
+            return sb.ToString();
+        }
+
+        public string GetSerializedValue(bool useNewValues = false)
+        {
+            var valueToSerialize = propertyValue;
+
+            if (useNewValues && newValue != null)
+            {
+                valueToSerialize = newValue;
+            }
+
+            if (valueToSerialize == null)
+                return "n";
```

Comment:
might be better to store these string constants into a const string, especially since they are also used in `DeserializeString()` below